tank_lab_to_nwb/convert_towers_task/towerspositiondatainterface.py

```diff
@@ -74,49 +74,37 @@ def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool =
         # TODO: move this to get_metadata in main converter
-        session_start_time = date_array_to_dt(matin['log']['session'][0][0]['start'][0][0][0])
+        session_start_time = array_to_dt(matin['log']['session']['start'])
         # session_start_time = metadata_dict['session_start_time']
 
         # Intervals
         if Path(mat_file).is_file():
             nwbfile.add_epoch_column('label', 'name of epoch')
 
-            blocks = matin['log']['block']
-            epoch_start_fields = blocks[0][0]['start'][0]
-            n_epochs = len(epoch_start_fields)
-            epoch_start_dts = [date_array_to_dt(x[0]) for x in epoch_start_fields]
-            epoch_durationss = [timedelta(seconds=x[0][0]) for x in blocks[0][0]['duration'][0]]
-
-            epoch_windows = []
+            n_epochs = len(matin['log']['block'])
+            epoch_start_dts = [array_to_dt(x.start) for x in matin['log']['block']]
+            epoch_durations = [timedelta(seconds=x.duration) for x in matin['log']['block']]
             for j in range(n_epochs):
                 start = epoch_start_dts[j] - session_start_time
-                end = start + epoch_durationss[j]
-                epoch_windows.append([start.total_seconds(), end.total_seconds()])
-
-            for j, window in enumerate(epoch_windows):
-                nwbfile.add_epoch(start_time=window[0], stop_time=window[1], label='Epoch'+str(j))
-
-            trial_start_fields = []
-            trial_duration_fields = []
-            n_trials = []
-            for j in range(n_epochs):
-                trial_start_fields.append(blocks[0][0]['trial'][0][j]['start'][0])
-                trial_duration_fields.append(blocks[0][0]['trial'][0][j]['duration'][0])
-                n_trials.append(len(trial_start_fields[j]))
-            trial_starts = [y[0][0]+epoch_windows[j][0] for j, x in enumerate(trial_start_fields) for y in x]
-            trial_ends = [y1[0][0]+epoch_windows[j][0]+y2[0][0]
-                          for j, x in enumerate(zip(trial_start_fields, trial_duration_fields))
-                          for y1, y2 in zip(x[0], x[1])]
-
+                end = start + epoch_durations[j]
+                nwbfile.add_epoch(
+                    start_time=start.total_seconds(),
+                    stop_time=end.total_seconds(),
+                    label='Epoch'+str(j+1)
+                )
+
+            trial_starts = [trial.start for epoch in matin['log']['block'] for trial in epoch.trial]
+            trial_durations = [trial.duration for epoch in matin['log']['block'] for trial in epoch.trial]
+            trial_ends = [start_time + duration for start_time, duration in zip(trial_starts, trial_durations)]
             for k in range(len(trial_starts)):
                 nwbfile.add_trial(start_time=trial_starts[k], stop_time=trial_ends[k])
 
             # Processed position
-            pos_obj = Position(name='_position')
+            pos_obj = Position(name="PositionSeries")
```

Comment:
@bendichter This is a placeholder name for now; what would a better one be? In the past this has been, for example, the name of the maze task the mouse is running during that epoch (if the maze changes per epoch, that is).

Also, there is only currently one PositionSeries that is all the concatenated processes position data, whereas in the past it has also been broken up per epoch. Should we keep it as one object here?

Reply:
It was broken up by epochs in the past because each epoch had a different environment. Here the environment is the same so the SpatialSeries should all be together